mimic/rest/maas_api.py

```diff
@@ -0,0 +1,960 @@
+
+class M_Cache(dict):
+
+    """
+    M(onitoring) Cache Object to hold dictionaries of all entities, checks and alarms.
+    """
+
+    def __init__(self):
+        """
+        Create the initial structs for cache
+        """
+        self.entities_list = []
+        self.checks_list = []
+        self.alarms_list = []
+        self.notifications_list = [{'id': 'ntTechnicalContactsEmail',
+                                    'label': 'Email All Technical Contacts',
+                                    'created_at': time.time(),
+                                    'updated_at': time.time(),
+                                    'metadata': None,
+                                    'type': 'technicalContactsEmail',
+                                    'details': None}]
+        self.notificationplans_list = [{'id': 'npTechnicalContactsEmail',
+                                        'label': 'Technical Contacts - Email',
+                                        'critical_state': [], 'warning_state': [],
+                                        'ok_state': [], 'metadata': None}]
+        self.notificationtypes_list = [{'id': 'webhook', 'fields': [{'name': 'url',
+                                                                     'optional': False,
+                                                                     'description': 'An HTTP or \
+                                                                      HTTPS URL to POST to'}]},
+                                       {'id': 'email', 'fields': [{'name': 'address',
+                                                                   'optional': False,
+                                                                   'description': 'Email \
+                                                                    address to send notifications to'}]},
+                                       {'id': 'pagerduty', 'fields': [{'name': 'service_key',
+                                                                       'optional': False,
+                                                                       'description': 'The PagerDuty \
+                                                                        service key to use.'}]},
+                                       {'id': 'sms', 'fields': [{'name': 'phone_number',
+                                                                 'optional': False,
+                                                                 'description': 'Phone number to send \
+                                                                  the notification to, \
+                                                                  with leading + and country \
+                                                                  code (E.164 format)'}]}]
+        self.suppressions_list = []
+
+
+def createEntity(params):
+    """
+    Returns a dictionary representing an entity
+    """
+    params = collections.defaultdict(lambda: '', params)
+    newentity = {}
```

Comment:
Minor style note: when you are building a dictionary like this, it's best to build it as a single value (as you do above) rather than one value at a time.  If you want to copy most of the stuff from `params` and make a few small tweaks, you can use `dict.update` to copy multiple values in one shot.